ote_sdk/ote_sdk/entities/dataset_item.py

```diff
@@ -276,13 +280,18 @@ def get_annotations(
 
                 shape_labels = annotation.get_labels(include_empty)
 
+                if not include_ignored:
+                    for label in shape_labels:
+                        if label.label in self.__ignored_labels:
+                            shape_labels.remove(label)
```

Comment:
You can't remove from a container while iterating on it. Use a comprehension or filter.